support/azureutil/azureutil.go

```diff
@@ -196,6 +196,11 @@ func IsAroHCP() bool {
 	return os.Getenv("MANAGED_SERVICE") == hyperv1.AroHCP
 }
 
+// IsSelfManagedAzure returns true if the managed service environment variable is set to something other than ARO-HCP or not set
+func IsSelfManagedAzure() bool {
+	return !IsAroHCP()
```

Comment:
this could be dangerous. I think it should take an HCP obj as an argument and ge something like